### PR TITLE
Implemented support for Terraform Enterprise as state provider

### PR DESCRIPTION
**Pull request by mvisonneau**
This MR will allow support of [Terraform Enterprise](https://www.terraform.io/docs/enterprise/index.html) (formerly Terraform Cloud) as the statefile provider. It will also make it easier to support additional providers.

### CODE REVIEW

File: state/state.go [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+// Provider is an interface for supported state providers
+type Provider interface {
+	GetLocks() (map[string]LockInfo, error)
+	GetVersions(string) ([]Version, error)
+	GetStates() ([]string, error)
+	GetState(string, string) (*statefile.File, error)
+}
+
+// Provider is a handler for the configured one
+var p Provider
```
Comment (raphink) [inline state/state.go]: I'd rather about global variables if possible. Could you have `Configure` return the provider instead?

File: state/tfe.go [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/camptocamp/terraboard/config"
+	tfe "github.com/hashicorp/go-tfe"
+	"github.com/hashicorp/terraform/states/statefile"
+	//log "github.com/sirupsen/logrus"
```
Comment (raphink) [inline state/tfe.go]: If it's unused, you might as well remove it

**Comment by raphink** [comment]
This is pretty nice, there's been several requests in the past for more providers.

File: state/tfe.go (hunk 2 above)
Comment (mvisonneau) [reply state/tfe.go]: indeed 🤦‍♂ 👍

File: state/state.go (hunk 1 above)
Comment (mvisonneau) [reply state/state.go]: Definitely, I refactored using locally scoped variables 👍

**Pull request merged**